js/stronghold.js

```diff
@@ -0,0 +1,459 @@
+        return {
+            'currency': code,
+            'address': this.checkAddress (address),
+            'info': response,
+        };
+    }
+
+    async withdraw (code, amount, address, tag = undefined, params = {}) {
+        await this.loadMarkets ();
+        let paymentMethod = undefined;
+        const currencyId = this.currencyId (code);
+        if (code in this.options['paymentMethods']) {
+            paymentMethod = this.options['paymentMethods'][code];
+        } else {
+            throw new InvalidArguments (this.id + ' fetchDepositAddress requires code to be BTC, ETH, or XLM');
+        }
+        const request = {
+            'assetId': currencyId,
+            'paymentMethod': paymentMethod,
+        };
+        const response = await this.privatePostVenuesVenueIdAccountsAccountIdDeposit (this.extend (request, params));
+        // {
+        //     "id": "5be48892-1b6e-4431-a3cf-34b38811e82c",
+        //     "assetId": "BTC/stronghold.co",
+        //     "amount": "10",
+        //     "feeAmount": "0.01",
+        //     "paymentMethod": "bitcoin",
+        //     "paymentMethodDetails": {
+        //       "withdrawal_address": "1vHysJeXYV6nqhroBaGi52QWFarbJ1dmQ"
+        //     },
+        //     "direction": "withdrawal",
+        //     "status": "pending"
+        //   }
+        return {
+            'id': response['result']['id'],
+            'info': response,
+        };
+    }
+
+    async fetchAccounts (params) {
+        return await this.privatePostVenuesVenueIdAccounts (params)
+    }
+
+    handleErrors (code, reason, url, method, headers, body, response) {
+        if (!response) {
+            return;
+        }
+        // { requestId: '3e7d17ab-b316-4721-b5aa-f7e6497eeab9',
+        //   timestamp: '2019-01-31T21:59:06.696855Z',
+        //   success: true,
+        //   statusCode: 200,
+        //   result: [] }
+        let errorCode = this.safeString (response, 'errorCode');
+        if (errorCode in this.exceptions) {
+            let Exception = this.exceptions[errorCode];
+            throw new Exception (this.id + ' ' + body);
+        }
+        const success = this.safeValue (response, 'success');
+        if (!success) {
+            throw new ExchangeError (this.id + ' ' + body);
+        }
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        const extractedParams = this.extractParams (path);
+        for (let i = 0; i < extractedParams.length; i++) {
+            let param = extractedParams[i];
+            if (!(param in params)) {
+                if (param === 'venueId') {
+                    params['venueId'] = this.options['venues']['main'];
+                } else if (param === 'accountId') {
+                    if (this.uid === undefined) {
+                        throw new AuthenticationError (this.id + ' requires uid');
```

Comment:
Let's not use this.uid for account ids, this.uid is for auth credentials only. Let's use `this.options['accountId']` for that instead.

Reply:
Also, I think we should not do the accountId logic in the `sign()` it's foreign to this place, we should mix it to the request in the unified methods.